### ISSUE TITLE
Add bicycle_parking=safe_loops to the bicycle parking quest

### ISSUE THREAD

**Issue by squeak9** [streetcomplete/StreetComplete#5774]
**Use case**
I feel like I see so many of these types of types of bicycle parking in Sweden but I thought they would just be classified as `bicycle_parking=wall_loops` because it's technically correct. I just recently looked into it and found out this more specific tag that's more specific: `bicycle_parking=safe_loops` which makes it more accurate.

It's like a wall loop but there is a thing on the side where you can lock the bike frame to the stand instead of the wheel which makes it safer.

Here is a nice image that's public domain: https://wiki.openstreetmap.org/wiki/File:Safe_loops_bicycle_parkint.png , more here: https://wiki.openstreetmap.org/wiki/Key:bicycle_parking
EDIT: more images:

- https://commons.wikimedia.org/wiki/File:Rastrelliera_bici_presso_Verona_11.jpg (CC BY-SA 4.0)
- https://commons.wikimedia.org/wiki/File:Fahrradst%C3%A4nder_mit_Anlehnb%C3%BCgel_2.jpg (CC0)
- https://commons.wikimedia.org/wiki/File:Fahrradst%C3%A4nder_mit_Anlehnb%C3%BCgel.jpg (CC0) **Best one**
- https://commons.wikimedia.org/wiki/File:Cykelf%C3%B6rr%C3%A5d_under_uppbyggnad_i_J%C3%A4rna.jpg (CC BY-SA 4.0)

**Proposed Solution**
Add it as an option to the bicycle parking type quest, 

You might say that the tag is rare but that might be because people didn't bother looking it up and just tagged it as `wall_loops`.

**Comment by mnalis** [comment]
**TL;DR:** it seems to be subcategory of `stands`, and is not hugely popular (`0.41%`). Probably better fit for SCEE, as it might be confusing in SC, where `bicycle_parking=stands` is currently defined as _"Stand (supports bicycle frame)"_

---

Some extra information:

- `safe_loops` has 2228 uses ([0.41% of `bicycle_parking=*` values](https://taginfo.openstreetmap.org/keys/bicycle_parking#values))
- Also see related  issue https://github.com/streetcomplete/StreetComplete/issues/1202 (and discussions linked there).

> but I thought they would just be classified as `bicycle_parking=wall_loops` because it's technically correct

Is it technically correct, though? Main characteristic of `bicycle_parking=wall_loops` is that the whole bicycle is kept upright **just by wheel being attached** to parking (resulting in popular name _"wheelbenders"_ for those parkings, as that is the damage they often cause due to ). It also has issues with securing a bike (as an U-lock will be able only to lock a wheel, allowing for the rest of a bike to be stolen).

So I would say that  **`bicycle_parking=safe_loops` is instead a sub-category of `bicycle_parking=stands`**, as the bike is secured from falling / **balanced by its frame** (instead of only by its wheel), **and both a wheel and a frame can be securely locked** preventing partial thefts. There are many other sub-categories of `stands`, see some examples [here](https://github.com/streetcomplete/StreetComplete/issues/1202#issuecomment-1972015739). Some have separate ([documented or not](https://taginfo.openstreetmap.org/keys/bicycle_parking#values)) OSM values, some don't - what is common for all of them for all is the parts I've **marked in bold**.

That being said, the `bicycle_parking=safe_loops` seems like acceptable category if one prefers to distinguish it (although I personally would prefer `bicycle_parking=stands` + `stands=safe_loops` method of specifying every-increasing level of detail, as it would avoid huge burden on data consumers[^1] and actually help increase the acceptance of the new tag!). 

It is just that if it was to be included in StreetComplete, **other answers should be redesigned** to include subtypes of parking along the types in order to be usable -- perhaps just change descriptions of more generic types, or better yet change it  to quest type like _"building types"_ which supports generic-types (like `residential`) as well as sub-types (like `house` or `apartments`).

[^1]: for example, if some data consumer (e.g. OsmAnd or cyclemap or whatever) does support that level on detail, with `bicycle_parking=safe_loops` it won't be able to show appropriately different icon or any type of information  except _"there is **some** bike parking here"_. It it were tagged as `bicycle_parking=stands` + `stands=safe_loops` instead, even if it did not know about `stands=safe_loops`  sub-tag, it would still be able to show the use that there are **Stands** there where user can safely lock their bicycle without damaging it, even if it could not say additional details (e.g. is it "safe loops" or "[wide_stands](https://wiki.openstreetmap.org/wiki/Tag:bicycle_parking%3Dwide_stands)" or "artistic" etc. subtype of "stands")

**Comment by mnalis** [comment]
> stands (German: Bügel) only attach the frame

I don't know about German wiki, but English wiki for [bicycle_parking=stands](https://wiki.openstreetmap.org/wiki/Tag:bicycle_parking=stands) explicitly states:

> A bent piece of metal against which you can lean your entire bicycle. Makes it **possible to lock the frame and a wheel to it**. Moderate security. Rectangular stands are sometimes called "staple racks" in North America due to their shape, and "Sheffield stands" in the UK. Use this tag for non-rectangular stands as well (e.g. round ones, fancy artistic ones, long ones allowing to attach more than two vehicles). 

(emphasis mine)

>  safe_loops attach rim and frame
> From my point of view, this is a huge 

I agree it would be big difference, if it were correct. However it seems to me that _"only attach the frame"_ for `bicycle_parking=stands` is **incorrect**. :man_shrugging: 

I'd say it is German translation problem then, as both pictures on the wiki  for `bicycle_parking=stands` (even on [German wiki](https://wiki.openstreetmap.org/wiki/DE:Key:bicycle_parking)) clearly show structures that allow **both the wheel and a frame** to be locked to it. It also seem supported by numbers (it seems quite unlikely that the [most popular](https://taginfo.openstreetmap.org/keys/bicycle_parking#values) type of bicycle_parking would allow locking **only** frame but **not** a wheel -- that would seem to me like extremely rare structure)

**Comment by matkoniecz** [comment]
> stands (German: Bügel) only attach the frame

why you think so? You definitely can attach also wheels on typical inverted U stand (though in areas with lesser amount of bicycle theft attaching only frame is popular)

https://www.stolenride.co.uk/resources/how-to-lock-your-bike/

https://www.rota.mt/bicycle-racks/

![screen02](https://github.com/user-attachments/assets/c5c57284-cc8f-4f22-8e60-6b9cdf35ef27)
![screen01](https://github.com/user-attachments/assets/b759ed68-4a7c-4142-ba76-ad1577049c27)

**Comment by matkoniecz** [comment]
You lean bicycle against stand, why it is not "supported standing"?

**Comment by mnalis** [comment]
@mcliquid thanks for the explanation above, it does make it more clear!

I agree that `safe_loops` has advantage over `wall_loops` in that in allows (in addition to wheel) the **frame to be locked**, which makes it significantly more theft-proof (while still not as good as `stands`; as judging by the pictures it only allows securing one wheel, not both)

However I am confused about parts on the wiki where you seem to imply that `safe_loops` would avoid "wheel bending"? If I understand you correctly above, the main advantage for carbon frames over `stands` is precisely that `safe_loops` **balances the bike _only_ on a wheel** (just as `wall_loops` do) and bike **frame does not lean / is not supported** by parking structure at all, right?

So any lateral force would still lean the whole of a bike to a side, putting all the pressure solely on the wheel rim? At least until the frame hit the structure, which would only happen in 50% of the cases commonly depending on the direction (For example in [picture](https://wiki.openstreetmap.org/wiki/File:Fahrradst%C3%A4nder_mit_Anlehnb%C3%BCgel_2.jpg) if force was pushing bike from left to right, the damage to the wheel would be limited as frame would hit the upper part, but if the force was pushing from right to left, the damage to the wheel could be as high as with regular `wall_loops`, right?)

P.S. (I'm not trying to argue here which bike parking is "better", but just trying to get a clear understanding of all the practical  differences, so they could be better documented)

**Comment by mnalis** [comment]
> That is correct and in my view this is also clear from the description? 

Thanks,  I'll try to improve wiki description later when I have access to regular computer again in about 2weeks..

> You can also see very clearly that no pressure/force is exerted on the rim itself on the centre bike.

it is not ***at the moment*** (when picture was taken), but it would be the same with regular wall_loops, no?

It is about what would happen ***if*** such force is applied (e.g. to take that same picture; if someone parking their bike in empty spot close to the leftmost bike were to accidentally push that leftmost bike hard to the left, its front wheel might get damaged, right?)

**Comment by mnalis** [comment]
**TL;DR:** Can the new value be added without introducing more confusion than the value it brings? I'm not sure :shrug:, but it does not look easy to me. Suggestions solicited.

---

> My point is just that there is a meaningful distinction between wall_loops, safe_loops and stands.

I absolutely agree with you on that, as well as that there is meaningful distinction between **all** other documented & used `bicycle_parking` values. If they weren't important to someone, people would not invent more detailed tag and use them... Yet SC only shows a subset of them (and there are reasons for that).

One question is thus _"is importance big enough to warrant separate SC answer"_. Now it obviously is important to you :smiley_cat:, and I _personally_ would like it. It is however a matter whether such answer is better match for [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) (where it would fit well, as requiring subject expertise is not a big issue), or _"is it easy enough answer to include for general StreetComplete population?"_ - which is IMHO more important second question?

---

Because, if it is included in SC, current answers for both `Stand` and `Wheelbender` (and not only the new `Safe loops`!) will likely need to change, and change in such a way that it is very clear to anyone (even persons not owning a bicycle!) which answer is correct one in each case.

I.e. currently user can visually exclude all "special" rack types (lower two rows) and only needs to distinguish between
- **Handlebar holder** (pretty easy to identify by its name alone)
- **Stands** (needs that "**supports bicycle frame**" to be understandable to most users)
- **Wheelbenders** (needs that "**supports wheel only**" to be understandable to most users)

So basically, most confusion to users is just between `stands` and `wall_loops`, and current explanation texts for those are (hopefully) clear enough to distinguish them. 

***Adding a third contender to the mix*** muddies up a situation a lot, and now ***requires three separate explanations how to clearly delineate all three them with using just 2-3 words*** (the picture will become invisible with more words), which seems like a hard problem (given most of target SC audience is not familiar with names of such bicycle parking types. Even some bicycle users might not be familiar with those names; and I don't think e.g. Croatian even _has_ a separate name for those).  

Or if more than 3 words are needed the quest would probably need to move to `Surface quality Quest` format, allowing for more text, but then again that would complicate the quest (although _hopefully_ not to the point of making it disabled by default).

Given above, do you @mcliquid  have suggestions how that problem should be approached? **What would be your suggested texts for `stands`, `wall_loops` &  `safe_loops`** (if the last one were to be added)?

![small_Screenshot_20240814_130820_StreetComplete](https://github.com/user-attachments/assets/ede5e313-7e5a-47ee-bede-7fe9ffa48814)

**Comment by westnordost** [comment]
So if you remove the wheelbenders from the safe loops, you end up with stands?

**Comment by mnalis** [comment]
> So if you remove the wheelbenders from the safe loops, you end up with stands? 

Judging by the pictures (included that last one in https://github.com/streetcomplete/StreetComplete/issues/5774#issuecomment-2288858645) I would say that if you were to saw-off the part holding the wheel from `safe_loops`, you would end up with something looking and functioning like _"regular"_ stands, **yes**. 
I don't know why @mcliquid seems to disagree here (if we've both understood the question in same way?)

>  (Why have the wheelbenders in that construction at all, then?)

One (only?) of the reasons mentioned in discussion was that **carbon frames are extremely sensitive to bumps with the metal** (which might case invisible microcracks making frame prone to catastrophic failures when later subjected to stress when riding technical track sections, IIUC). And I guess even owners of many fancy aluminum/steel frames owners might dislike paint being scraped off from their new bicycle (which is likely when parking a bicycle on inverted-U type stands). 

Having the bicycle **_balanced by the wheel only_** is in such cases **lesser of the two evils** (truing the wheel or even buying new one is faster and cheaper than a new paint job, and certainly better then buying a new carbon frame or being involved in an accident). And some models of safe loops might even reduce the extensive damage to the wheel (by transferring damage from the wheel to the frame instead in there is too much lateral motion - see 4th paragraph of https://github.com/streetcomplete/StreetComplete/issues/5774#issuecomment-2260335448)

**Comment by westnordost** [comment]
I didn't want to add stuff that requires new translations in v59.x while it was already near-release. I think the implementation as contributed to SCEE by @mcliquid makes sense.

**Comment by westnordost** [comment]
But anyway, sorry still that it took so long.

**Comment by squeak9** [comment]
No problem, thanks for adding it! 👍
